#174797455 Setup-Redux

Adadie opened the pull request:
#### What does this PR do?
This PR sets up redux in the project.

#### Description of Task to be completed?
- install redux.
- install react-redux.
- add redux devtool extension.
- create a store.
- setup reducers folder.

#### How should this be manually tested?
- Clone this repo `https://github.com/atlp-rwanda/script-heroes-frontend.git`.
- Check out this branch `ch-setup-redux-#174797455`.
- Install the redux dev tool in your browser, follow this link to download it if [you use chrome](https://chrome.google.com/webstore/detail/redux-devtools/lmhkpmbekcpmknklioeibfkpmmfibljd).
- Run `yarn`.
- Run `yarn start`.
- Inspect the home with redux, the tree should be empty because we don't have any states yet.

#### Any more relevant info?

> The warning in the console: `Store does not have a valid reducer. Make sure the argument passed to combineReducers is an object whose values are reducers` occurs because there is no state in the reducers files.

#### What are the relevant pivotal tracker stories?
[#174797455](https://www.pivotaltracker.com/story/show/174797455)
#### Screenshots (if appropriate)
<img width="1438" alt="Screen Shot 2020-11-12 at 12 46 56" src="https://user-images.githubusercontent.com/68109866/98931431-ae488b80-24e6-11eb-9445-d5c69445673a.png">

dushimeemma commented on src/index.js:
```diff
@@ -1,6 +1,19 @@
 import React from "react";
```
please separate redux store with index.js file and also try to use `thunk` and `applymiddleware`

b0nbon1 commented:
Could you kindly rebase and resolve the conflicts?

dushimeemma commented on redux/store.js:
```diff
@@ -0,0 +1,15 @@
+
+const store = createStore(
+  allReducers,
+  applyMiddleware(...middlewares),
```
please wrap applyMiddleware and redux devtools in compose

Habinezajanvier committed:
chore(front-end): set redux for state management

- set thunk for asynchronise state management
- display states using redux dev tools

[finishes #174797455]
